pkg/auth/oauth.go

```diff
@@ -101,8 +102,13 @@ func (m *OAuthMiddleware) Wrap(next http.Handler) http.Handler {
 		// Check for token in cookie
 		cookie, err := r.Cookie(m.cookieName)
 		if err == nil && cookie.Value != "" {
-			// Validate token (you could add caching here if needed)
-			if m.validateToken(cookie.Value) {
+			if user, err := m.getUser(cookie.Value); err == nil {
+				pr := new(http.Request)
+				*pr = *r
```

Comment:
Not modifying inplace here, [because](https://pkg.go.dev/net/http?utm_source=godoc#Handler)

> Except for reading the body, handlers should not modify the provided Request.